src/batch.jl

```diff
@@ -169,8 +168,8 @@ end
         # @show tid, ThreadingUtilities._atomic_state(tid)
         ThreadingUtilities.wait(tid)
       end
-      free_threads!(torelease)
     end
+    free_threads!(torelease_tuple)
```

Comment:
This'll need an update to the Project.toml (to ensure we have a new enough PolyesterWeave).
You can wait until we've released a version of PolyesterWeave with your fixes.